nixos/modules/services/networking/unbound.nix

```diff
@@ -211,9 +281,29 @@ in
         RestrictNamespaces = true;
         LockPersonality = true;
         RestrictSUIDSGID = true;
+
+        Restart = "always";
```

Comment:
Is the default restart policy not enough anymore? Did you run into a higher failure rate?

(Note: I'm not against this just do not see the *why* this was made here or in the commit message)

Reply:
I probably just forgot to remove it.

Reply:
However without it, tests don't pass anymore :/

Reply:
That does not sound correct. The services shouldn't have to be restarted multiple times.

Reply:
```
authoritative # [   12.742517] unbound[771]: [1606080723] unbound[771:0] error: can't bind socket: Cannot assign requested address for fd21::1 port 53
authoritative # [   12.745242] unbound[771]: [1606080723] unbound[771:0] fatal error: could not open ports
authoritative # [   12.753705] systemd[1]: unbound.service: Main process exited, code=exited, status=1/FAILURE
authoritative # [   12.760274] systemd[1]: unbound.service: Failed with result 'exit-code'.
authoritative # [   12.767436] systemd[1]: Failed to start Unbound recursive Domain Name Server.
authoritative # [   12.774673] systemd[1]: Reached target Host and Network Name Lookups.[   12.872069] 8021q: adding VLAN 0 to HW filter on device eth0
authoritative # 
authoritative # [   12.779442] dhcpcd[622]: eth0: waiting for carrier
authoritative # [   12.782018] dhcpcd[622]: eth0: carrier acquired
authoritative # [   12.803541] dhcpcd[622]: DUID 00:01:00:01:27:4d:95:53:52:54:00:12:34:56
authoritative # [   12.805385] dhcpcd[622]: eth0: IAID 00:12:34:56
authoritative # [   12.806970] dhcpcd[622]: eth0: adding address fe80::5054:ff:fe12:3456
```
According to the logs, `fd21::1` was already added before.

Reply:
I'll have to look at this in detail. We should ship with a configuration that doesn't require daemons to restart during normal startup..

Reply:
I didn't have a look into this but as long as this is here that is a huge -1 from me. We should ship stable services by default.

Reply:
From what I can tell on my setup, it doesn't restart during startup. But I'd still want to have that `Restart=always` in there, in case it fails during operation.

Reply:
It would be better to make those errors appear and are fixed than just restarting a service. At least we must downgrade this to `on-failure` as otherwise the service will be restarted if it terminates with exit 0.

Also you can always add that to your local configuration if you desire that behavior.

Reply:
Updated to use `on-failure`.